File: backend/src/main/java/br/ufpr/sept/so2/modules/presenca/api/dto/ConfirmarPresencaRequest.java
package br.ufpr.sept.so2.modules.presenca.api.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;

public record ConfirmarPresencaRequest(
        @NotBlank String pin,
        @NotBlank @Size(min = 8, max = 64) String deviceUuid,
        @NotBlank String fase
) {
}

File: backend/src/main/kotlin/br/ufpr/sept/so2/modules/presenca/api/dto/ConfirmarPresencaRequest.kt
package br.ufpr.sept.so2.modules.presenca.api.dto

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Size

data class ConfirmarPresencaRequest(
    val pin: String? = null,
    val token: String? = null,
    @field:NotBlank @field:Size(min = 8, max = 64) val deviceUuid: String,
    @field:NotBlank val fase: String,
)
